core/embed/rust/src/ui/geometry.rs

```diff
@@ -503,40 +539,45 @@ pub struct LinearPlacement {
 }
 
 impl LinearPlacement {
-    pub fn horizontal() -> Self {
+    pub const fn horizontal() -> Self {
         Self {
             axis: Axis::Horizontal,
             align: Alignment::Start,
             spacing: 0,
         }
     }
 
-    pub fn vertical() -> Self {
+    pub const fn vertical() -> Self {
         Self {
             axis: Axis::Vertical,
             align: Alignment::Start,
             spacing: 0,
         }
     }
 
-    pub fn align_at_start(mut self) -> Self {
-        self.align = Alignment::Start;
-        self
+    pub const fn align_at_start(self) -> Self {
+        Self {
+            align: Alignment::Start,
+            ..self
+        }
     }
 
-    pub fn align_at_center(mut self) -> Self {
-        self.align = Alignment::Center;
-        self
+    pub const fn align_at_center(self) -> Self {
+        Self {
+            align: Alignment::Center,
+            ..self
+        }
     }
 
-    pub fn align_at_end(mut self) -> Self {
-        self.align = Alignment::End;
-        self
+    pub fn align_at_end(self) -> Self {
```

Comment:
const missing, also in Rect::split_right

Reply:
fixed in ab2ba1947